homeassistant/components/sensor/netdata.py

```diff
@@ -4,154 +4,152 @@
+import logging
 
-import requests
 import voluptuous as vol
 
-import homeassistant.helpers.config_validation as cv
 from homeassistant.components.sensor import PLATFORM_SCHEMA
 from homeassistant.const import (
-    CONF_HOST, CONF_PORT, CONF_NAME, CONF_RESOURCES)
+    CONF_HOST, CONF_ICON, CONF_NAME, CONF_PORT, CONF_RESOURCES)
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['netdata==0.1.2']
 
 _LOGGER = logging.getLogger(__name__)
-_RESOURCE = 'api/v1'
-_REALTIME = 'before=0&after=-1&options=seconds'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
+
+CONF_ELEMENT = 'element'
 
 DEFAULT_HOST = 'localhost'
 DEFAULT_NAME = 'Netdata'
-DEFAULT_PORT = '19999'
-
-SCAN_INTERVAL = timedelta(minutes=1)
-
-SENSOR_TYPES = {
-    'memory_free': ['RAM Free', 'MiB', 'system.ram', 'free', 1],
-    'memory_used': ['RAM Used', 'MiB', 'system.ram', 'used', 1],
-    'memory_cached': ['RAM Cached', 'MiB', 'system.ram', 'cached', 1],
-    'memory_buffers': ['RAM Buffers', 'MiB', 'system.ram', 'buffers', 1],
-    'swap_free': ['Swap Free', 'MiB', 'system.swap', 'free', 1],
-    'swap_used': ['Swap Used', 'MiB', 'system.swap', 'used', 1],
-    'processes_running': ['Processes Running', 'Count', 'system.processes',
-                          'running', 0],
-    'processes_blocked': ['Processes Blocked', 'Count', 'system.processes',
-                          'blocked', 0],
-    'system_load': ['System Load', '15 min', 'system.load', 'load15', 2],
-    'system_io_in': ['System IO In', 'Count', 'system.io', 'in', 0],
-    'system_io_out': ['System IO Out', 'Count', 'system.io', 'out', 0],
-    'ipv4_in': ['IPv4 In', 'kb/s', 'system.ipv4', 'received', 0],
-    'ipv4_out': ['IPv4 Out', 'kb/s', 'system.ipv4', 'sent', 0],
-    'disk_free': ['Disk Free', 'GiB', 'disk_space._', 'avail', 2],
-    'cpu_iowait': ['CPU IOWait', '%', 'system.cpu', 'iowait', 1],
-    'cpu_user': ['CPU User', '%', 'system.cpu', 'user', 1],
-    'cpu_system': ['CPU System', '%', 'system.cpu', 'system', 1],
-    'cpu_softirq': ['CPU SoftIRQ', '%', 'system.cpu', 'softirq', 1],
-    'cpu_guest': ['CPU Guest', '%', 'system.cpu', 'guest', 1],
-    'uptime': ['Uptime', 's', 'system.uptime', 'uptime', 0],
-    'packets_received': ['Packets Received', 'packets/s', 'ipv4.packets',
-                         'received', 0],
-    'packets_sent': ['Packets Sent', 'packets/s', 'ipv4.packets',
-                     'sent', 0],
-    'connections': ['Active Connections', 'Count',
-                    'netfilter.conntrack_sockets', 'connections', 0]
-}
+DEFAULT_PORT = 19999
+
+DEFAULT_ICON = 'mdi:desktop-classic'
+
+RESOURCE_SCHEMA = vol.Any({
+    vol.Required(CONF_ELEMENT): cv.string,
+    vol.Optional(CONF_ICON, default=DEFAULT_ICON): cv.icon,
```

Comment:
Isn't icon part of customize already?

Reply:
It's but this avoid that the configuration has to be split in multiple sections.

Reply:
I want to get rid of customize. It's an ugly hack. So I approve of this 👍